ehrql/query_engines/base_sql.py

```diff
@@ -414,13 +417,21 @@ def to_first_of_month(self, date):
 
     @get_sql.register(Function.MaximumOf)
     def get_sql_maximum_of(self, node):
+        aggregate_function = self.aggregate_functions()["maximum_of"]
         args = self.get_nary_function_args(node)
-        return self.get_aggregate_subquery(sqlalchemy.func.max, *args).label("greatest")
+        return self.get_aggregate_subquery(aggregate_function, *args).label("greatest")
 
     @get_sql.register(Function.MinimumOf)
     def get_sql_minimum_of(self, node):
+        aggregate_function = self.aggregate_functions()["minimum_of"]
         args = self.get_nary_function_args(node)
-        return self.get_aggregate_subquery(sqlalchemy.func.min, *args).label("least")
+        return self.get_aggregate_subquery(aggregate_function, *args).label("least")
+
+    def aggregate_functions(self):
```

Comment:
I'd be inclined to make the 'base' functions the canonical ANSI SQL functions for this which are `greatest` and `least`

Reply:
As to what goes in the base engine and what goes in the child engines, I don't think we've really got a consistent story yet (or certainly not one that's consistently applied).

I've had a vague idea of trying to do it along the lines Jon suggests ("standard" stuff goes in the base engine) but I don't think I've really followed that in practice.

I'm fine with just deciding on pragmatic grounds for now and we can always rework this stuff later without too much difficulty if we want to.